File: 2962-Count-Subarrays-Where-Max-Element-Appears-at-Least-K-Times.js
var countSubarrays = function(nums, k) {
    let maxVal = Math.max(...nums);
    let numSubArrays = 0;
    let currentCount = 0;
    let j = 0;

    for (let i = 0; i < nums.length; i++) {
        if (nums[i] === maxVal) {
            currentCount++;
        }
        while (currentCount >= k) {
            if (nums[j] === maxVal) {
                currentCount--;
            }
            j++;
        }
        numSubArrays += j;
    }

    return numSubArrays;
};

File: 2962-Count-Subarrays-Where-Max-Element-Appears-at-Least-K-Times.py
class Solution(object):
    def countSubarrays(self, nums, k):
        maxVal = max(nums)
        numSubArrays = 0
        currentCount = 0
        j = 0
        for i in range(len(nums)):
            currentCount += (1 if nums[i] == maxVal else 0)
            while currentCount >= k:
                currentCount -= (1 if nums[j] == maxVal else 0)
                j += 1
            numSubArrays += j

        return numSubArrays
